### ISSUE TITLE
Is it compatible with cakephp 4.4.7 ?

### ISSUE THREAD

**Issue by rudy1976s** [burzum/cakephp-service-layer#33]
I am trying to implement the plugin on cake version 4.4.7 and I found several incompatibilities.
Does this version is compatible with recent cakephp ?

Thank you

**Comment by dereuromark** [comment]
Yes, it should. I am using it on latest Cake - see also running tests (if you run them).
Whats the issue - you need to be a bit more specific here.

I see some issues around paginator changes so far only.

**Comment by dereuromark** [comment]
AHA, looks like CakePHP released a BC break by not having those interfaces inherit from each other?
Or the class alias doesnt work as expected in type hints?
```php
class_alias(
    'Cake\Datasource\Paging\PaginatorInterface',
    'Cake\Datasource\PaginatorInterface'
);
deprecationWarning(
    'Use Cake\Datasource\Paging\PaginatorInterface instead of Cake\Datasource\PaginatorInterface.'
);
```

=> https://github.com/cakephp/cakephp/issues/16867

**Comment by rudy1976s** [comment]
Hi Mark! that's the error!!!

**Comment by rudy1976s** [comment]
In the code there is the "old fashion" $this->loadModel as well, which seems deprecated, as far as I know

**Comment by dereuromark** [comment]
Deprecation is not a BC issue, though.

**Comment by dereuromark** [comment]
After https://github.com/cakephp/cakephp/commit/88a4f70d6339e50daa2b382b8af1c1be6b976388 this will be released as fix in 4.4.8 :)

**Comment by rudy1976s** [comment]
Thank you very much Mark!
